File: 김영재/6주차/명예의 전당.js
function solution(k, score) {
    return score.map((_, idx) => {
        return score.slice(0, +idx + 1).sort((a, b) => b - a).slice(0,k).sort((a, b) => a - b)[0]
    });
    //배열을 순회하며 1일차 까지, 2일차 까지, ...n일차 까지로 잘라서 그 날짜의 점수들 중 상위 K개만 잘라와서 가장 작은 점수를 반환
}

console.log(solution(3, [10, 100, 20, 150, 1, 100, 200]));
console.log([10, 100, 20, 150, 1, 100, 200].slice(0, 1));

File: 김영지/24주차/java/명예의전당.java
import java.util.Arrays;
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (int i = 0; i < score.length; i++) {
            pq.offer(score[i]);

            if (pq.size() > k) {
                pq.poll();
            }

            answer[i] = pq.peek();
        }

        return answer;
    }
}

public class 명예의전당 {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] result = solution.solution(3, new int[] {10, 100, 20, 150, 1, 100, 200});

        System.out.println(Arrays.toString(result));
    }
}
